src/Symfony/Component/HttpKernel/HttpCache/Psr6Store.php

```diff
@@ -0,0 +1,616 @@
+    {
+        if ($response->headers->has('X-Content-Digest')) {
+            return;
+        }
+
+        $contentDigest = $this->generateContentDigest($response);
+
+        if (null === $contentDigest) {
+            return;
+        }
+
+        $digestCacheItem = $this->cache->getItem($contentDigest);
+
+        if ($digestCacheItem->isHit()) {
+            $cacheValue = $digestCacheItem->get();
+
+            // BC
+            if (\is_string($cacheValue)) {
+                $cacheValue = [
+                    'expires' => 0, // Forces update to the new format
+                    'contents' => $cacheValue,
+                ];
+            }
+        } else {
+            $cacheValue = [
+                'expires' => 0, // Forces storing the new entry
+                'contents' => $this->isBinaryFileResponseContentDigest($contentDigest) ?
+                    $response->getFile()->getPathname() :
+                    $response->getContent(),
+            ];
+        }
+
+        $responseMaxAge = (int) $response->getMaxAge();
+
+        // Update expires key and save the entry if required
+        if ($responseMaxAge > $cacheValue['expires']) {
+            $cacheValue['expires'] = $responseMaxAge;
+
+            if (false === $this->saveDeferred($digestCacheItem, $cacheValue, $responseMaxAge)) {
+                throw new \RuntimeException('Unable to store the entity.');
+            }
+        }
+
+        $response->headers->set('X-Content-Digest', $contentDigest);
+
+        // Make sure the content-length header is present
+        if (!$response->headers->has('Transfer-Encoding')) {
+            $response->headers->set('Content-Length', \strlen((string) $response->getContent()));
+        }
+    }
+
+    /**
+     * Test whether a given digest identifies a BinaryFileResponse.
+     *
+     * @param string $digest
+     */
+    private function isBinaryFileResponseContentDigest($digest): bool
+    {
+        return 'bf' === substr($digest, 0, 2);
+    }
+
+    /**
+     * Increases a counter every time a write action is performed and then
+     * prunes expired cache entries if a configurable threshold is reached.
+     * This only happens during write operations so cache retrieval is not
+     * slowed down.
+     */
+    private function autoPruneExpiredEntries(): void
+    {
+        if (0 === $this->options['prune_threshold']) {
+            return;
+        }
+
+        $item = $this->cache->getItem(self::COUNTER_KEY);
+        $counter = (int) $item->get();
+
+        if ($counter > $this->options['prune_threshold']) {
+            $this->prune();
+            $counter = 0;
+        } else {
+            ++$counter;
+        }
+
+        $item->set($counter);
+
+        $this->cache->saveDeferred($item);
+    }
+
+    /**
+     * @param int   $expiresAfter
+     * @param array $tags
+     */
+    private function saveDeferred(CacheItemInterface $item, $data, $expiresAfter = null, $tags = []): bool
```

Comment:
```suggestion
    private function saveDeferred(CacheItemInterface $item, $data, int $expiresAfter = null, array $tags = []): bool
```
and remove the doc block?